### PR TITLE
improve algorithm documentation [skip ci]

### LINKED ISSUE #1564: Update README and other documentation to better describe algorithm status
The README just lists all algorithms at the same level.  We should clarify which algorithms have been selected for NIST standardization, which are in Round 4 or at other standards bodies like ISO, which algorithms are highly experimental (once we start adding signature on-ramp submissions); which build options will enable which algorithms, etc.

### PR DESCRIPTION
**Pull request by baentsch**
Fixes #1564 : @dstebila please check whether this is enough to close out the remaining blocker for the release of 0.10.0. A full resolution may entail introducing a new algorithm class "STD-track" [as discussed elsewhere](https://github.com/open-quantum-safe/liboqs/pull/1716#issuecomment-1976252544) (leaving only "ML-KEM" and "ML-DSA" in the STD designation). But IMO this warrants some more deliberation and should not stop 0.10.0.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,7 +40,11 @@ The OQS project is supported by the [Post-Quantum Cryptography Alliance](https:/
...
+The list below indicates all algorithms currently supported by liboqs, including experimental algorithms and already excluding algorithm variants pruned during the NIST competition, such as Kyber-90s or Dilithium-AES.
+
+The currently only algorithms selected for standardization are the "ML-KEM" and "ML-DSA" variants with their respective different bit strengths. `liboqs` will retain these algorithm names selected by NIST throughout the finishing stages of the standardization process, so users can rely on their presence going forward. If NIST changes the implementation details of these algorithms, `liboqs` will adapt the implementation such as for users to not be impacted by such potential change. For users interested in explicitly selecting the current "proposed draft standard" code, the variants with the suffix "-ipd" are made available. At this stage, "ml-kem-ipd" and "ml-kem" as well as "ml-dsa-ipd" and "ml-dsa" are functionally equivalent, denoted by the "alias" moniker below.
```
Comment (SWilson4) [inline README.md:45]:
```suggestion
The only algorithms currently selected for standardization are the "ML-KEM" and "ML-DSA" variants with their respective different bit strengths. `liboqs` will retain these algorithm names selected by NIST throughout the finishing stages of the standardization process, so users can rely on their presence going forward. If NIST changes the implementation details of these algorithms, `liboqs` will adapt the implementation such as for users to not be impacted by such potential change. For users interested in explicitly selecting the current "proposed draft standard" code, the variants with the suffix "-ipd" are made available. At this stage, "ml-kem-ipd" and "ml-kem" as well as "ml-dsa-ipd" and "ml-dsa" are functionally equivalent, denoted by the "alias" moniker below.
```
Looks like a couple of words flip-flopped.

**Review by SWilson4 — APPROVED** [review]
LGTM barring one typo. Thanks to [The Incredible Baentsch](https://github.com/bcgit/bc-csharp/discussions/527#discussioncomment-8699203) for putting this together. :stuck_out_tongue_winking_eye:

File: README.md [hunk 2]
```diff
@@ -40,7 +40,11 @@ The OQS project is supported by the [Post-Quantum Cryptography Alliance](https:/
...
+The list below indicates all algorithms currently supported by liboqs, including experimental algorithms and already excluding algorithm variants pruned during the NIST competition, such as Kyber-90s or Dilithium-AES.
+
+The only algorithms currently selected for standardization are the "ML-KEM" and "ML-DSA" variants with their respective different bit strengths. `liboqs` will retain these algorithm names selected by NIST throughout the finishing stages of the standardization process, so users can rely on their presence going forward. If NIST changes the implementation details of these algorithms, `liboqs` will adapt the implementation such as for users to not be impacted by such potential change. For users interested in explicitly selecting the current "proposed draft standard" code, the variants with the suffix "-ipd" are made available. At this stage, "ml-kem-ipd" and "ml-kem" as well as "ml-dsa-ipd" and "ml-dsa" are functionally equivalent, denoted by the "alias" moniker below.
```
Comment (dstebila) [inline README.md:45]: The first sentence in this paragraph is a bit confusing to me. Falcon and SPHINCS+ have also been selected by NIST for standardization.

Comment (baentsch) [reply README.md:45]: > Falcon and SPHINCS+ have also been selected by NIST for standardization.

OK, I stand corrected -- I thought those names and parameter sets are not yet final. Is there a doc like FIPS 203 to point to?

Comment (baentsch) [reply README.md]: Changed as suggested. Please re-check and approve @dstebila to close the last issue out for 0.10.0.

Comment (SWilson4) [reply README.md]: Falcon and SPHINCS+ have both been selected for standardization, but those names will change. For Falcon, there is no FIPS draft yet. For SPHINCS+, [there is](https://csrc.nist.gov/pubs/fips/205/ipd), but we don't yet have an implementation of "SLH-DSA" (and indeed I'm not sure what tweaks the draft makes to the Round 3 version of SPHINCS+).

I think it makes sense to reword this to state something like
```
The only algorithms in `liboqs` that implement NIST standards drafts are the "ML-KEM" and "ML-DSA" variants...
```
and additionally add a sentence saying
```
Falcon and SPHINCS+ have also been selected for standardization, but the `liboqs` implementations of these algorithms are currently tracking Round 3 submissions and not NIST standards drafts.
```
I think this conveys the commitment to keep the ML-KEM and ML-DSA names (and currently only those names) stable throughout the standardization process, while recognizing that Falcon and SPHINCS+ are also soon to be standardized. Whenever we have implementations of SLH-DSA and NTRU-DSA (or whatever other name NIST settles on for Falcon) we can add them to this list of stables names too.

Sorry, I should have caught this on my first pass.

Comment (dstebila) [reply README.md]: That sounds great Spencer, go for it.

File: README.md [hunk 3]
```diff
@@ -40,7 +40,11 @@ The OQS project is supported by the [Post-Quantum Cryptography Alliance](https:/
...
+The list below indicates all algorithms currently supported by liboqs, including experimental algorithms and already excluding algorithm variants pruned during the NIST competition, such as Kyber-90s or Dilithium-AES.
+
+The only algorithms currently selected for standardization are the "ML-KEM", "ML-DSA", "Falcon" and "Sphincs+" variants with their respective different bit strengths. `liboqs` will retain these algorithm names selected by NIST throughout the finishing stages of the standardization process, so users can rely on their presence going forward. If NIST changes the implementation details of these algorithms, `liboqs` will adapt the implementation such as for users to not be impacted by such potential change. For users interested in explicitly selecting the current "proposed draft standard" code, the variants with the suffix "-ipd" are made available. At this stage, "ml-kem-ipd" and "ml-kem" as well as "ml-dsa-ipd" and "ml-dsa" are functionally equivalent, denoted by the "alias" moniker below.
```
Comment (SWilson4) [inline README.md:45]:
```suggestion
The only algorithms in `liboqs` that implement NIST standards drafts are the [`ML-KEM`](https://csrc.nist.gov/pubs/fips/203/ipd) and [`ML-DSA`](https://csrc.nist.gov/pubs/fips/204/ipd) variants with their respective different bit strengths. `liboqs` will retain these algorithm names selected by NIST throughout the finishing stages of the standardization process, so users can rely on their presence going forward. If NIST changes the implementation details of these algorithms, `liboqs` will adapt the implementation such as for users to not be impacted by such potential change. For users interested in explicitly selecting the current "proposed draft standard" code, the variants with the suffix "-ipd" are made available. At this stage, "ml-kem-ipd" and "ml-kem" as well as "ml-dsa-ipd" and "ml-dsa" are functionally equivalent, denoted by the "alias" moniker below.

Falcon and SPHINCS+ have also been [selected for standardization](https://csrc.nist.gov/Projects/post-quantum-cryptography/selected-algorithms-2022), but the `liboqs` implementations of these algorithms are currently tracking Round 3 submissions and not NIST standards drafts.
```

Adding as a suggestion to avoid potentially introducing a conflict on @baentsch's branch.

File: README.md [hunk 4]
```diff
@@ -40,7 +40,13 @@ The OQS project is supported by the [Post-Quantum Cryptography Alliance](https:/
...
+
+Falcon and SPHINCS+ have also been [selected for standardization](https://csrc.nist.gov/Projects/post-quantum-cryptography/selected-algorithms-2022), but the `liboqs` implementations of these algorithms are currently tracking Round 3 submissions and not NIST standards drafts.
+
+All other algorithm names are subject to change. `liboqs` makes available a [selection mechanism for algorithms on the NIST standards track, continued NIST competition, or purely experimental nature by way of the configuration variable OQS_ALGS_ENABLED](CONFIGURE.md#oqs_algs_enabled). By default `liboqs` is built supporting all, incl. experimental, PQ algorithms listed below.
```
Comment (baentsch) [inline README.md:49]: Re-reading, I wonder whether the term "other" is now still clear/appropriate? Falcon and Sphincs (names) will change...

Comment (SWilson4) [reply README.md:49]: All names other than ML-KEM and ML-DSA would perhaps be more clear.

File: README.md (hunk 2 above)
Comment (SWilson4) [reply README.md]: Marking this conversation as resolved, as this change has now been applied.

**Review by dstebila — APPROVED** [review]
